From b57b79a7a070932c3b13627b09aae69580f5215b Mon Sep 17 00:00:00 2001
From: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Date: Mon, 1 Jun 2026 10:50:28 +0200
Subject: [PATCH 2/4] feat(st): add Linux data partition GUID

The Generic Linux Data Partition is a standard defined by the
Discoverable Partitions Specification (DPS) to identify generic Linux
file systems containing data [1].

Defining this GUID (0fc63daf-8483-4772-8e79-3d69d8477de4) allows TF-A
to correctly identify these partitions.

[1] https://uapi-group.org/specifications/specs/discoverable_partitions_specification/#:~:text=Generic%20Linux%20Data%20Partition
Change-Id: Icc274223c818c6df3f39cab7556c99bb21bb18ba
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Upstream-Status: Backport [https://git.trustedfirmware.org/plugins/gitiles/TF-A/trusted-firmware-a/+/caf7a47b57e1227b9c28af84ff4097b072d2440a]
---
 plat/st/common/include/stm32mp_efi.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/common/include/stm32mp_efi.h b/plat/st/common/include/stm32mp_efi.h
index 63e1bd86da03..c12ec6f9d803 100644
--- a/plat/st/common/include/stm32mp_efi.h
+++ b/plat/st/common/include/stm32mp_efi.h
@@ -9,6 +9,10 @@
 
 #include <drivers/partition/efi.h>
 
+#define LINUX_FILE_SYSTEM_DATA_GUID \
+	EFI_GUID(0x0FC63DAFU, 0x8483U, 0x4772U, \
+		 0x8EU, 0x79U, 0x3DU, 0x69U, 0xD8U, 0x47U, 0x7DU, 0xE4U)
+
 #define STM32MP_FIP_GUID \
 	EFI_GUID(0x19d5df83U, 0x11b0U, 0x457bU, \
 		 0xbeU, 0x2cU, 0x75U, 0x59U, 0xc1U, 0x31U, 0x42U, 0xa5U)
-- 
2.43.0

